### PR TITLE
Feature/cost forecast fetch

### PR DESCRIPTION
**Pull request by LisaQ04**
Added Costs Forecast actions and reducers for daily and monthly.

### CODE REVIEW

File: waterlog/src/Components/DailyCostsReports.js [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
 
-const DailyCostsReports = (props) => { 
-  let labelX = props.props.dataPoints.map(a => (new Date(a.x).getHours() + ":00"));
-  let dataY = props.props.dataPoints.map(a => Math.round(a.y)); 
+const DailyCostsReports = (props) => {
+  let labelX = props.props.dailyCost.dataPoints.map(a => (new Date(a.x).getHours() + ":00"));
+  let dataY = props.props.dailyCost.dataPoints.map(a => Math.round(a.y)); 
+  const {yIntercept, slope} = props.props.forecastDaily[0];
+
+  const startX = Math.floor((new Date(props.props.dailyCost.dataPoints[0].x).getTime()) / 1000);
```
Comment (RashadAkoodieRabbit) [inline waterlog/src/Components/DailyCostsReports.js]: Consider null checking when accessing an index on your array

Comment (FredJEhlers) [inline waterlog/src/Components/DailyCostsReports.js]: Please extract the `new Date(props.props.dailyCost.dataPoints[0].x).getTime()` and `new Date(props.props.dailyCost.dataPoints[labelX.length-1].x).getTime()` into variables

File: waterlog/src/Components/Usage.js [hunk 2]
```diff
@@ -33,101 +33,81 @@ class Usage extends Component {
...
-            <button
-              className={`btn-graph-nav tablinks ${
-                this.state.display === "daily" ? "active" : ""
-              }`}
-              onClick={e => this.openGraph("daily")}
-              id="openByDefault"
-            >
-              Daily
-            </button>
+      <div className="wastage">
```
Comment (RashadAkoodieRabbit) [inline waterlog/src/Components/Usage.js]: Please leave spaces between your "=". It improves readibility

File: waterlog/src/Components/Usage.js [hunk 3]
```diff
@@ -33,101 +33,81 @@ class Usage extends Component {
...
-          <MonthlyCostsReports props={this.props.monthlyCost} />
-          <hr id="divide" />
-          <p className="bold">Usage</p>
-          <MonthlyUsageComponent props={this.props.monthlyUsage} />
-        </div>
-      );
+      return <div>
+        <p className="bold">Cost</p>
+        <MonthlyCostsReports props={this.props} /> 
+        <hr id="divide"/>
```
Comment (FredJEhlers) [inline waterlog/src/Components/Usage.js]: Ids are supposed to be unique, so if you are reusing it, then make use of classes

File: waterlog/src/Components/Usage.js [hunk 4]
```diff
@@ -33,101 +33,81 @@ class Usage extends Component {
...
   render() {
-    const {
-      dailyError,
-      dailyLoading,
-      monthlyError,
-      monthlyLoading,
-      seasonalError,
-      seasonalLoading
-    } = this.props;
+    const { dailyError, dailyLoading,
```
Comment (FredJEhlers) [inline waterlog/src/Components/Usage.js]: each on its own line

File: waterlog/src/Components/Usage.js [hunk 5]
```diff
@@ -33,101 +33,81 @@ class Usage extends Component {
...
-              className={`btn-graph-nav tablinks ${
-                this.state.display === "daily" ? "active" : ""
-              }`}
-              onClick={e => this.openGraph("daily")}
-              id="openByDefault"
-            >
-              Daily
-            </button>
+      <div className="wastage">
+        <div className="graph-nav tab " id="cost-buttons">
```
Comment (FredJEhlers) [inline waterlog/src/Components/Usage.js]: inconsistent naming scheme for ids

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: waterlog/src/actions/CostsMonthlyForecastAction.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+});
+
+export const fetchCostsForecastMonthlyFailure = error => ({
+  type: FETCH_COSTS_FORECAST_FAILURE_MONTHLY,
+  payload: {
+    error
+  }
+});
+async function getMonthlyForecast(dispatch) {
+  let arr = []
```
Comment (RashadAkoodieRabbit) [inline waterlog/src/actions/CostsMonthlyForecastAction.js]: array of what? Better naming please

File: waterlog/src/actions/CostsMonthlyForecastAction.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  for (let i = 1; i <= 12; i++) {
+    let f = await getMonthForecast(i).then(seg => arr.push(seg)).catch(error => {
```
Comment (RashadAkoodieRabbit) [inline waterlog/src/actions/CostsMonthlyForecastAction.js]: Where are you using f?

**Review by RashadAkoodieRabbit — CHANGES_REQUESTED** [review]

File: waterlog/src/Components/MonthlyCostsReports.js [hunk 8]
```diff
@@ -2,57 +2,67 @@ import React from 'react'
...
+          fill: true,
+          borderColor: 'rgba(255,0,0,0)',
+          backgroundColor: 'rgba(255,0,0,1)'
+        },
+        {
+          label: 'forecast',
+          data: forecast,
+          fill: true,
+          borderColor: 'rgb(0,191,255)',
+          backgroundColor: 'rgb(0,191,255)'
```
Comment (CilliersN) [inline waterlog/src/Components/MonthlyCostsReports.js]: line 15-16 is rgba
stay with it. 
change 23-23 to it

**Review by CilliersN — CHANGES_REQUESTED** [review]

File: waterlog/src/Components/DailyCostsReports.js [hunk 9]
```diff
@@ -16,7 +26,17 @@ const DailyCostsReports = (props) => {
         pointBackgroundColor: 'rgba(255,23,68,1)',
         pointRadius: 5,
         pointHitRadius: 5
+      },
+      {
+        label: 'forecast',
+        data: forecast,
+        fill: true,
+        borderColor: 'rgb(0,191,255)', 
+        pointBackgroundColor: 'rgb(0,191,255)',
```
Comment (CilliersN) [inline waterlog/src/Components/DailyCostsReports.js]: rgba

Comment (LisaQ04) [reply waterlog/src/Components/DailyCostsReports.js]: Now Using rgba consistently.

File: waterlog/src/Components/MonthlyCostsReports.js (hunk 8 above)
Comment (LisaQ04) [reply waterlog/src/Components/MonthlyCostsReports.js]: changed to rgba.

File: waterlog/src/actions/CostsMonthlyForecastAction.js (hunk 6 above)
Comment (LisaQ04) [reply waterlog/src/actions/CostsMonthlyForecastAction.js]: Changed array naming.

File: waterlog/src/Components/Usage.js (hunk 2 above)
Comment (LisaQ04) [reply waterlog/src/Components/Usage.js]: Done

File: waterlog/src/Components/Usage.js (hunk 3 above)
Comment (LisaQ04) [reply waterlog/src/Components/Usage.js]: Changed to class name.

File: waterlog/src/Components/Usage.js (hunk 4 above)
Comment (LisaQ04) [reply waterlog/src/Components/Usage.js]: Each on its own line.

File: waterlog/src/Components/DailyCostsReports.js (hunk 1 above)
Comment (LisaQ04) [reply waterlog/src/Components/DailyCostsReports.js]: Moved into variables.

File: waterlog/src/actions/CostsMonthlyForecastAction.js (hunk 7 above)
Comment (LisaQ04) [reply waterlog/src/actions/CostsMonthlyForecastAction.js]: removed unnecessary variable.

**Review by RashadAkoodieRabbit — APPROVED** [review]

**Review by CilliersN — APPROVED** [review]

**Review by FredJEhlers — APPROVED** [review]

**Pull request merged**